File: Crio/crio_programming_interview_problems-master/FactorialTrailingZeroes/FactorialTrailingZeroes.cpp
#include <bits/stdc++.h>
#include "FastIO.hpp"

using namespace std;

class FactorialTrailingZeroes {
public:
    int trailingZeroes(int n) {
        int ans = 0;
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        for(long int i = 5; i <= n; i*=5) {
            ans += n/i;
        }
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
        return ans;
    }
};

int main() {
	FastIO();
	int n;
	cin >> n;
	int result = FactorialTrailingZeroes().trailingZeroes(n);
	cout << result;
	return 0;
}

File: Crio/crio_programming_interview_problems-master/FactorialTrailingZeroes/FactorialTrailingZeroes.java
import java.io.*;
import java.util.*;

public class FactorialTrailingZeroes {
    // Implement your solution by completing the below function
    public int trailingZeroes(int n) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        if (n / 5 == 0)
            return 0;
        return n / 5 + trailingZeroes(n / 5);
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

        // CRIO_UNCOMMENT_START_MODULE_L1_PROBLEMS
        //return -1;
        // CRIO_UNCOMMENT_END_MODULE_L1_PROBLEMS
    }

    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);
        int number = scanner.nextInt();
        scanner.close();
        int result = new FactorialTrailingZeroes().trailingZeroes(number);
        System.out.print(result);
    }
}
